docs/x509/ocsp.rst

```diff
@@ -285,6 +286,55 @@ Creating Responses
             :class:`~cryptography.x509.ReasonFlags` enumeration or ``None`` if
             the ``cert`` is not revoked.
 
+    .. method:: add_response_by_hash(issuer_name_hash, issuer_key_hash, serial_number, algorithm, cert_status, this_update, next_update, revocation_time, revocation_reason)
+
+        .. versionadded:: 45.0.0
+
+        This method adds status information about the certificate that was
+        requested to the response using the hash values directly, rather than requiring
+        the full certificates. You can call this method or ``add_response``
+        only once.
+
+        :param issuer_name_hash: The hash of the issuer's DER encoded name using the
+            same hash algorithm as the one specified in the ``algorithm`` parameter.
+        :type issuer_name_hash: bytes
+
+        :param issuer_key_hash: The hash of the issuer's public key bit string
+            DER encoding using the same hash algorithm as the one specified in
+            the ``algorithm`` parameter.
+        :type issuer_key_hash: bytes
+
+        :param serial_number: The serial number of the certificate being checked.
+        :type serial_number: int
+
+        :param algorithm: A
+            :class:`~cryptography.hazmat.primitives.hashes.HashAlgorithm`
+            instance. For OCSP only
+            :class:`~cryptography.hazmat.primitives.hashes.SHA1`,
+            :class:`~cryptography.hazmat.primitives.hashes.SHA224`,
+            :class:`~cryptography.hazmat.primitives.hashes.SHA256`,
+            :class:`~cryptography.hazmat.primitives.hashes.SHA384`, and
+            :class:`~cryptography.hazmat.primitives.hashes.SHA512` are allowed.
+
+        :param cert_status: An item from the
+            :class:`~cryptography.x509.ocsp.OCSPCertStatus` enumeration.
+
+        :param this_update: A naïve :class:`datetime.datetime` object
```

Comment:
Do we want to be doing new naïve APIs?

Reply:
I'm actually a bit unsure. Our choices are to make it inconsistent with `add_response`, or to make it inconsistent with how we want APIs to work in the future :-/

Reply:
Should we add add_response_utc, deprecate add_response, and then just name the new one add_response_by_hash_utc? Although that latter name is pretty incoherent.